change: Track sprite atlas textures for updating

FlashyReese opened the pull request:
Sodium currently tracks sprites from `SpriteIdentifier` which uses `SpriteAtlasTexture`. The `SpriteIdentifier` is not used for every case such as painting. This PR moves the texture tracking to `SpriteAtlasTexture` this means we no longer need to track from `SpriteIdentifier`.

- Fixes #248
- Should also Fix #403 

This PR was tested with https://github.com/FlashyReese/sodium-fabric/commit/ce2db9c292f90d289709ff469f087d606be24061 and can be tested with #526

FlashyReese committed:
change: Track sprite atlas holder textures

jellysquid3 commented:
Maybe [this commit](https://github.com/CaffeineMC/sodium-fabric/commit/342402ea4e6fdef4e138bcdaccd7411600f747c5) should instead just be cherry-picked? It covered more cases in my testing.

FlashyReese commented:
If you are referring to the removed changes, if your commit you mentioned allowing for mods to render their own entity fire esque effects. That commit has made it in the current branch.

FlashyReese committed:
change: Track sprite atlas textures for updating

FlashyReese commented:
PaintingManager does not use SpriteIdentifier, hence animated paintings are not covered.

FlashyReese committed:
change: Don't track textures from SpriteIdentifier.

We no longer need to track from `SpriteIdentifier`, as we are already tracking from the sprite atlas.

FlashyReese committed:
Terrible naming on my behalf
